src/lib/README.md

```diff
@@ -0,0 +1,202 @@
+## Specialized data structures
+
+### Ctable (lib.ctable)
+
+A *ctable* is a hash table whose keys and values are instances of FFI
+data types.  In Lua parlance, an FFI value is a "cdata" value, hence the
```

Comment:
I quite like the introduction, including both *use* and *how*, but maybe I'm more interested in the details of the *how* than the average user.